iothub/device/samples/convention-based-samples/Thermostat/ThermostatSample.cs

```diff
@@ -79,15 +79,21 @@ private async Task HandlePropertyUpdatesAsync(ClientPropertyCollection writableP
                         _logger.LogDebug($"Property: Received - [ \"{targetTemperatureProperty}\": {targetTemperatureRequested}°C ].");
 
                         _temperature = targetTemperatureRequested;
+                        /*
+                        feedback: this is an odd place to serialize.  
+                        Why do we serialize here instead of doing this inside UpdateClientPropertiesAsync? 
+                        Do we ask the customer to serialize anything else?  
+                        */
+
```

Comment:
Originally we had an Add(value, status, version, ...) that would either create a generic class that was serializable by the two proposed serailizers (Newtonsoft and Sys.Text.JSON) when the collection was used before the client was inializsed. Or, it would create a concrete type that used the Serializer annotations.

Reply:
The general feedback was that Add(value, status, version, ...) was confusing to use since all other "Add" methods take in component name, property name and property value, while this method took in a bunch of additional params.
On having the "Add" method always take in just component name, property name and value, this was the trade-off - the customer either needs to know what type of IWritableResponse to create (NewtonSoftWritableResponse, their own serializer based writable response, etc), or they could use the payload convention information available under device/module client to know what their IWritableResponse should look like.

The main point here is that the IWritableResponse object needs to have the metadata annotations based on the serializer you use.